GeneralsMD/Code/Tools/wdump/chunk_d.cpp

```diff
@@ -1052,6 +1052,65 @@ void	ChunkTableClass::List_W3D_CHUNK_PS2_SHADERS(ChunkItem * Item,CListCtrl *Lis
 	}
 }
 
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADERS(ChunkItem * Item,CListCtrl *List)
+{
+	List_Subitems(Item,List);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER(ChunkItem* Item, CListCtrl* List)
+{
+	List_Subitems(Item, List);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER_INFO(ChunkItem* Item, CListCtrl* List)
+{
+	uint8 version = *(uint8*)Item->Data;
+	W3dFXShaderInfoStruct* shaderinfo = (W3dFXShaderInfoStruct*)((uint8*)Item->Data + 1);
+	int counter = 0;
+
+	AddItem(List, counter, "Version", version);
+	AddItem(List, counter, "ShaderName", shaderinfo->ShaderName);
+	AddItem(List, counter, "Technique", shaderinfo->Technique);
+}
+
+void	ChunkTableClass::List_W3D_CHUNK_FX_SHADER_CONSTANT(ChunkItem* Item, CListCtrl* List)
+{
+	int counter = 0;
+	uint8* chunkdata = (uint8 *)Item->Data;
+	uint32 type = *(uint32*)chunkdata;
+	uint32 strlen = *(uint32*)(chunkdata + 4);
```

Comment:
Perhaps choose a var name that is not `strlen`, because `strlen` is a popular C function.